### PR TITLE
address potential flake in GithubAppCredentialsTest

### PR DESCRIPTION
**Pull request by jtnord**
calling setLabels on an agent will not persist the node.

in older versions of Jenkins the tests would be less flaky as adding any
Node would cause all labels to be re-evaluated, so when creating a few
agents and adding labels in a loop the last one created would at least
deterministically ensure that all previous agents labels where correct.

However since 2.332 (jenkinsci/jenkins#5882)
only labels part of a node added or removed would be updated, and when
creating the agents they where created without labels, which where added
later.

This has the potential to cause tests to be flaky depending on when the periodic
trimLabels was called (or at least on other timing related things)

Additionally creating agents and waiting for them to come oneline is
slow. A job can be scheduled and the pipeline will run and then wait for the node to be
available,
so we can do other things whilst the agent is connecting.

# Description

A brief summary describing the changes in this pull request. See 
[JENKINS-XXXXX](https://issues.jenkins-ci.org/browse/JENKINS-XXXXX) for further information. 

# Submitter checklist
- [ ] Link to JIRA ticket in description, if appropriate.
- [ ] Change is code complete and matches issue description
- [ ] Automated tests have been added to exercise the changes
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verify that the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

# Documentation changes
- [ ] Link to jenkins.io PR, or an explanation for why no doc changes are needed

# Users/aliases to notify

### CODE REVIEW

**Review by jglick — COMMENTED** [review]
Is https://github.com/jenkinsci/github-branch-source-plugin/blob/d4d268cc4e1d504efdf3aea953d50c09b529757c/src/test/java/org/jenkinsci/plugins/github_branch_source/GithubAppCredentialsTest.java#L551 safe in case the agent has not yet connected?

File: src/test/java/org/jenkinsci/plugins/github_branch_source/GithubAppCredentialsTest.java [hunk 1]
```diff
@@ -112,8 +112,7 @@ public static void setUpJenkins() throws Exception {
     store.addCredentials(Domain.global(), appCredentialsNoOwner);
 
     // Add agent
-    agent = r.createOnlineSlave();
-    agent.setLabelString("my-agent");
+    agent = r.createSlave("my-agent", null);
```
Comment (jtnord) [inline src/test/java/org/jenkinsci/plugins/github_branch_source/GithubAppCredentialsTest.java:115]:
```suggestion
    agent = r.createSlave("my-agent", null);
    r.waitOnline(agent);
```

**Review by jglick — APPROVED** [review]

**Comment by jglick** [comment]
Hmm, seems to fail now.

**Comment by jtnord** [comment]
> Hmm, seems to fail now.

Yup,  how ironic... should not have been caused by this change (may always have been slightly flakey) will need some more investigation, so moving to draft.

**Comment by jtnord** [comment]
obsolete

**Pull request closed without merging**